File: pipes.js
class Pipe{
  constructor(){
    this.spacing = 125;
    this.y_top = random(height / 6, 3 / 4 * height);
    this.y_bot = this.y_top + this.spacing;
    this.x = width;
    this.width = 20;
    this.velocity = 3;
  }
  placePipe(){
    fill(255);
    rect(this.x,0,this.width,this.y_top);
    rect(this.x,this.y_bot,this.width,this.y_bot + height);
  }
  update() {
    this.x -= this.velocity;
  }
  hits(bird){
    if(bird.x >= this.x){
       if((bird.y > this.y_bot) || (bird.y < this.y_top)){
        return true;
      }
    }
    return false;
  }
}

File: python_nn/pipes.py
import pygame
import random
import main


class Pipe:
    def __init__(self,canvas_width,canvas_height):
        self.spacing = 100
        self.y_top = random.uniform(canvas_height / 6, 3/4 * canvas_height)
        self.y_bot = self.y_top + self.spacing
        self.x = canvas_width
        self.height = canvas_height
        self.width = 60
        self.velocity = 3
        self.top_pipe = None
        self.bottom_pipe = None

    def placePipe(self,canvas):
        self.top_pipe = pygame.draw.rect(canvas,main.white,(self.x,0,self.width,
                                                           self.y_top))
        self.bottom_pipe = pygame.draw.rect(canvas,main.white,(self.x,self.y_bot,
                                                              self.width,
                                            self.y_bot+self.height))
    def update(self):
        self.x -= self.velocity

    def hit(self,bird):
        if(self.x < bird.x + bird.size):
            return self.top_pipe.collidepoint((bird.x,bird.y)) | self.bottom_pipe.collidepoint((
                bird.x,bird.y))
        else:
            return False
